### PR TITLE
cdap-2397 Fixed bug in stopping of the Workflow.

### PR DESCRIPTION
**Pull request by sagarkapare**
Jira: https://issues.cask.co/browse/CDAP-2397

Fixed to catch the `InterruptedException` when the Workflow is stopped explicitly. This was done for the fork, but should also be done for the Workflow which does not contain the fork.

### CODE REVIEW

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-RBT258-1

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java [hunk 1]
```diff
@@ -401,54 +416,116 @@ public void testWorkflowForkApp() throws Exception {
...
 
+    // Get the currently running RunRecord for the Workflow
     List<RunRecord> historyRuns = getProgramRuns(programId, "running");
     Assert.assertTrue(historyRuns.size() == 1);
+    RunRecord record = historyRuns.get(0);
+    String runId = record.getPid();
 
-    String runId = historyRuns.get(0).getPid();
+    // Wait till first action in the Workflow starts executing
+    while (!firstSimpleActionFile.exists()) {
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: Incase of failure we will never come out of the loop. There should be a max retry and throw exception if the file doesn't exist after the retry

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java [hunk 2]
```diff
@@ -401,54 +416,116 @@ public void testWorkflowForkApp() throws Exception {
...
+    historyRuns = getProgramRuns(programId, "running");
+    Assert.assertTrue(historyRuns.size() == 1);
+    record = historyRuns.get(0);
+    Assert.assertTrue(!runId.equals(record.getPid()));
+
+    // Store the new RunId
+    runId = record.getPid();
+
+    // Wait till first action in the Workflow starts executing
+    while (!firstSimpleActionFile.exists()) {
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: Same here.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java [hunk 3]
```diff
@@ -401,54 +416,116 @@ public void testWorkflowForkApp() throws Exception {
...
+      TimeUnit.MILLISECONDS.sleep(50);
+    }
+    verifyRunningProgramCount(programId, runId, 1);
+
+    // Signal the first action to continue
+    firstSimpleActionDoneFile.createNewFile();
+
+    // Wait till fork in the Workflow starts executing
+    while (!(oneSimpleActionFile.exists() && anotherSimpleActionFile.exists())) {
       TimeUnit.MILLISECONDS.sleep(50);
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: also here.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java [hunk 4]
```diff
@@ -401,54 +416,116 @@ public void testWorkflowForkApp() throws Exception {
...
     waitState(programId, "RUNNING");
 
+    // Store the new RunRecord for the currently running run
     historyRuns = getProgramRuns(programId, "running");
     Assert.assertTrue(historyRuns.size() == 1);
     runId = historyRuns.get(0).getPid();
 
-    while (!(oneActionFile.exists() && anotherActionFile.exists())) {
+    // Wait till first action in the Workflow starts executing
+    while (!firstSimpleActionFile.exists()) {
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: one more.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: Fixed to add the retry logic.

**Comment by sagarkapare** [comment]
@sreevatsanraman Please take a look again. Thanks!

**Comment by sagarkapare** [comment]
New build running here - http://builds.cask.co/browse/CDAP-RBT258-2

**Comment by sreevatsanraman** [comment]
LGTM :+1:  Please merge when the build passes.

**Comment by sreevatsanraman** [comment]
Build Passed. Merging.

**Pull request merged**